topostats/plotting.py

```diff
@@ -51,6 +51,13 @@ def create_parser() -> arg.ArgumentParser:
         required=False,
         help="Filename to write a sample YAML configuration file to (should end in '.yaml').",
     )
+    parser.add_argument(
+        "--create-label-file",
```

Comment:
This is an excellent feature if I understand it correctly, allowing custom labelling of plots?

It does not currently work though, as the config is expecting a `base_dir`.

```
❯ toposum --create-label-file labels.yaml
[Wed, 01 Mar 2023 14:56:39] [INFO    ] [topostats] [plotting] Default configuration file loaded.
[Wed, 01 Mar 2023 14:56:39] [INFO    ] [topostats] [plotting] Default variable to labels mapping loaded.
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniforge/base/bin/toposum", line 8, in <module>
    sys.exit(main())
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/topostats/plotting.py", line 354, in main
    write_yaml(
  File "/opt/homebrew/Caskroom/miniforge/base/lib/python3.10/site-packages/topostats/io.py", line 66, in write_yaml
    config["base_dir"] = str(config["base_dir"])
KeyError: 'base_dir'
```

I presume we would want to add a `base_dir` to `summary_config.yaml` instead of changing `write_yaml` to allow a lack of `base_dir`, though both are a little messy...

In `io.py`, in `write_yaml`, line 65:
```python
    if "base_dir" in config.keys():
        config["base_dir"] = str(config["base_dir"])
```

or:

`plotting.py`, `main`, line 340:
```python
        config['base_dir'] = config['output_dir']
```

Some documentation would be great to add too.

Reply:
You've caught me tying myself up in :knot: as I first had `base_dir` in `topostats/summary_config.yaml` then removed it then decided it was more sensible to add it back in and not have `write_yaml()` check for its presence. I'll sorted out, thanks for picking it up.